### ISSUE TITLE
VS test runner process fails to exit if Pester script imports a module with a binary dependency

### ISSUE THREAD

**Issue by MatthewMWR** [pester/Pester#620]
For some projects-- vstest.executionengine process fails to exit after processing Pester tests, leaving files locked and projects unable to subsequently build.

I'm still working to minimize the repro steps, but as of now the closest I have is:

1. Have a C# class library project like My.CoolThing containing SomeCode.cs
2. In this project have a My.CoolThing.psd1 and psm1 exposing the functionality of My.CoolThing.dll to powershell callers
3. The psd1 has RequiredAssemblies = .\My.CoolThing.dll
4. In this project also have a My.CoolThing.Tests.Ps1 which calls Import-Module for My.CoolThing module and then does any kind of test Describe/Context/It
5. Uncheck Test > Test Settings > Keep Test Execution Engine Running
6. Run test via Test Explorer

Expected Result:
7. After running tests, vstest.executionengine process should exit, thereby releasing My.CoolThing.dll so future builds succeed.
Actual Result:
8. vstest.executionengine process persists forever, causing problems with future builds, CI, or what have you. Also, if you choose "Debug Test" instead of "Run Test" in Test Explorer, the debugging session never ends.

**Comment by nohwnd** [comment]
Could you check that this triggers the same issue?
```powershell
 $dll = '.\My.CoolThing.dll'
 Add-Type -TypeDefinition "namespace  My.CoolThing { public class SomeCode { } }" -OutputAssembly $dll
 "running cool module"| Set-Content .\MyCoolModule.psm1
 New-ModuleManifest -Path .\MyCoolModule.psd1 -RequiredAssemblies $dll
 Import-Module .\MyCoolModule.psd1

 Describe "assembly is loaded" {
    It "is loaded" {
        [appdomain]::currentdomain.getassemblies()  | where {$_.Location -like "*cool*"} | Should Not BeNullOrEmpty
    }
 }
```
Seems like the problem is with the test runner rather than pester/powershell itself, becuse once assembly is loaded to an appdomain, it cannot be unloaded without closing the app domain. That's how .NET always worked, and it's no different in PowerShell. On the other hand, it's weird that this would show up now, after the test running has been in the wild for multiple years. So please check if my code triggers the issue and then I can look into it further.

**Comment by MatthewMWR** [comment]
Hi! I have discovered that the issue was specific to the dll in question rather than Pester. For some reason whenever this DLL is loaded, something goes wrong with unloading the runspace and processes do not exit when expected.
For example it wasn't just vstest.executionengine.exe that would fail to exit, but powershell_ise would also fail to exit completely if the DLL had been loaded (UI would close, but process would remain).
Closing as this is not specific to Pester.
